File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/data/remote/DriverSessionReconcile.kt
package com.pegasusx.driver.data.remote

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/** Refetch server-authoritative driver snapshots after transport reconnect. */
suspend fun reconcileDriverSession(api: DriverApi) {
    runCatching { api.getAssignedOrders() }
    val today = SimpleDateFormat("yyyy-MM-dd", Locale.US).format(Date())
    runCatching { api.getManifest(today) }
}

File: pegasusX/apps/driver-app-ios/driverappios/driverappios/Services/DriverSessionReconcile.swift
//
//  DriverSessionReconcile.swift
//  driverappios
//
//  Refetch server-authoritative driver snapshots after transport reconnect.
//

import Foundation

enum DriverSessionReconcile {
    static func run() async {
        _ = try? await APIClient.shared.getAssignedOrders()
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd"
        formatter.locale = Locale(identifier: "en_US_POSIX")
        let today = formatter.string(from: Date())
        _ = try? await APIClient.shared.getManifest(date: today)
    }
}
